### ISSUE TITLE
FR: handling 2 or more of the same monitor

### ISSUE THREAD

**Issue by nielm** [birnenlabs/chrome-extensions#1]
Love the extension, but I have an issue: 

If you have more than 2 of the same monitor, it is difficult to specify the monitor to use in the actions: for example 2 external screens connected to a chromebook show up as: 

```
name: 'Built-in display', primary: 'true', internal: 'true'
name: 'HP Z27n', primary: 'false', internal: 'false'
name: 'HP Z27n', primary: 'false', internal: 'false'
```
So it is impossible to specify which of the 2 HP Z27N monitors to use. 

Suggestions

1) Allow matching on display.id  --- This unique ID seems to be persistent across reconnections and reboots, but I do not know _how_ persistent it actually is. 
2) Allow matching on the N'th instance of the matching name: eg: `display: "HP Z27N[0]` or `display: "HP Z27N[1]` --- The problem here is that it relies on ChromeOS keeping the displays in the same order in the results.
3) Allow matching on multiple attributes, eg `display: "HP Z27N,primary"` could work if you set one of the external displays to primary, but this only works for 2 displays.
4) Allow matching on the bounds of the display within the desktop: 
    * `displayBounds: {top: 0, left:0}` would match the display at the top left,
    * `displayBounds: {top: 0, left: 2560}` would match the display at the top right.

Let me know what you think, I am happy to work on a PR to implement.

**Comment by nielm** [comment]
(draft PR for option 4: https://github.com/birnenlabs/chrome-extensions/pull/2)

**Comment by lukaszg84** [comment]
What would you say about matching by monitor id? It could reuse the same display field (it is now handled in a somewhat special way, by using negation and primary or internal names) - it can try match against display name and id - I think it is unlikely to have collisions.

**Comment by nielm** [comment]
> What would you say about matching by monitor id? It could reuse the same display field (it is now handled in a somewhat special way, by using negation and primary or internal names) - it can try match against display name and id - I think it is unlikely to have collisions.

That's option 1. 

As mentioned, I am concerned about how persistent the [Chrome.system.display.id](https://developer.chrome.com/docs/extensions/reference/api/system/display) actually is...  

I will monitor the values on my laptop for a week and see if they change at all -- they seem to be large random integers, (ex 0x1e6d29968f6d02)

**Comment by lukaszg84** [comment]
I think I saw another extension that was using the ids. I decided to go for a display name as I didn't consider the same monitors.

I actually have another idea that is option 2 and 4 merged.

I am not a fan of option 4 as it requires setting a very specific values for a display, and they can change when you change dpi.

What can be done is to index displays but according to the order from the top left corner.
The monitors will be sorted by the left attribute (and if two monitors have the same left value they will be compared by their top) and then they can be indexed.
This way it will easily match multiple screen arrangements (e.g. when at work you have two HPs and at home there are 2 Dells).

For four monitors the sorting order would be the following.
```
  1
0 2 3
```

This can be even used with `-internal` matcher (`-internal[0]` and `-internal[1]`) to match any two external monitors.

**Comment by nielm** [comment]
So I have done a lot of playing, and it seems that the chrome.system.display.id _is_ pretty persistent. 

I tried connecting displays differently using different adaptors, different combinations of external displays, different connectors (HDMI and DVI), and they kept the same ID. 

I have a draft PR in #3 which handles this, but I also went down a bit of a rabbit hole cleaning up some things, adding validation to the Options page and removing some runtime errors. Let me know if you want me to separate this PR into 2 different ones -- one for cleanup/bug fixes and the other for the new ID matching feature. 

The suggestion of suffixing by sort order/desktop position is also interesting -- in PR #3 I sort the displays by bounds.left then bounds.top when displaying them in the options, so this is easily possible...  Definitely one for a separate PR though!
